src/OpenSage.Game/Gui/Apt/ActionScript/InstructionCollection.cs

```diff
@@ -309,8 +288,38 @@ public static InstructionCollection Parse(Stream input, long instructionsPositio
                             break;
                         case InstructionType.SetRegister:
                             instruction = new SetRegister();
-                            parameters.Add(Value.FromInteger(reader.ReadInt32()));
+                            parameters.Add(Value.FromRegister(reader.ReadUInt32()));
+                            break;
+                        case InstructionType.GotoLabel:
+                            instruction = new GotoLabel();
+                            parameters.Add(Value.FromString(reader.ReadStringAtOffset()));
+                            break;
+                        case InstructionType.PushData: // NIE doubtful and not used in any games
```

Comment:
This instruction is used by the [Battle for Middleearth](https://en.wikipedia.org/wiki/The_Lord_of_the_Rings:_The_Battle_for_Middle-earth) games, which were the first SAGE games that used Apt for UI. Please make sure you stay compatible in that regard.

Reply:
I see. I just added a line of throwing an exception to remind me where it is used. Although I'm mainly developing with RA3's scripts, I'll check the compatibility of BFME.